lua/telekasten.lua

```diff
@@ -203,6 +206,33 @@ local function escape(s)
     return string.gsub(s, "[%%%]%^%-$().[*+?]", "%%%1")
 end
 
+local function recursive_substitution(dir, old, new)
+    if not global_dir_check() then
+        return
+    end
+
+    if vim.fn.executable("find") == 0 then
+        vim.api.nvim_err_write("Find not installed!\n")
+        return
+    end
+    if vim.fn.executable("sed") == 0 then
+        vim.api.nvim_err_write("Sed not installed!\n")
+        return
+    end
+
+    os.execute(
```

Comment:
I completely overlooked that. Please check out plenary.job - that might make your issues regarding stdout go away.

```lua
local Job = require("plenary.job")
Job:new({
    command = "find",
    args = { dir, "-type", "f", "-name", ...},
    -- ...
   }):sync()
```
Another alternative (that possibly wouldn't solve anything) would be to make a shell script, put it in a scripts dir and execute it. M.base_directory .. "/scripts/foo.sh"` would be your script's name.

What I don't get is your sed syntax. I always use `s/a/b/g` - no `|`. Is there a difference?

Reply:
> Please check out plenary.job - that might make your issues regarding stdout go away.

Ok I will do that.

> What I don't get is your sed syntax. I always use s/a/b/g - no |. Is there a difference?

They are equivalent technically, the difference is that when using `|`, it is no longer required to escape `/` (as we might have in filepath here). It is a lazy workaround, I admit... I have sanitized all inputs using the escape_char list from telescope, so it is no longer necessary.

Reply:
I just found [nvim-spectre](https://github.com/nvim-pack/nvim-spectre), which seems to be doing what I want to do (search and replace across unopened files using `sed` and with `pleanry.job`). But it is well beyond my skill level and I am not sure how I should proceed...

Reply:
I haven't looked at it yet. But my take is: it's always good to have options. Often they might overlap significantly. 

It depends: we could mention nvim-spectre in the docs as a proposed solution to mass-renaming. Or we might roll our own here. Clearly, as with everything, the devil is in the details. So there might be something super wiki / telekasten specific that nvim-spectre just cannot know about. Like that `[[daily/xxxxxx]]` might belong to an entirely different directory when `daily /` is not part of `home`. Taking care of special cases like that is something only telekasten can do, I suppose.

Let me know what you think. There's no need to completely over-engineer something. Often, the braindead simple things work best - especially when everybody can estimate their limits. I have the impression that often people happily work around super simple solutions, not expecting them to be able to handle the super-complicated  ones of their use-cases. However, the moment you come up with a more sophisticated solution, everybody and their dog expect their super-specific workflow/configuration/setup to be fully supported, backed by 1000 lines of code if necessary :smile:.